src/http/conn.rs

```diff
@@ -155,7 +155,8 @@ where I: Io,
     }
 
     fn maybe_park_read(&mut self) {
-        if self.io.poll_read().is_ready() {
+        // TODO: this needs fixing
```

Comment:
d'oh right, sorry I thought I had handled this elsewhere already.

I was wondering if you know how the best way might be to handle this off-hand?